### PR TITLE
fix(code-snippet): remove focus from snippet container

### PR DESCRIPTION
**Pull request by tw15egan**
Refs https://github.com/carbon-design-system/carbon/issues/2304

Removes focus outline from the main text part of the Code Snippet (MultiLine only). The focus will now go directly to the copy button, then show more.

#### Changelog

**New**

- Reset focus styles on the code snippet container when multiline

**Removed**

- `tabindex` and `role` on multiline code snippet

#### Testing / Reviewing

Tab through / click on `CodeSnippet` and ensure no focus is given to the container. The focus should move to the copy button.

### CODE REVIEW

File: packages/components/src/components/code-snippet/_code-snippet.scss [hunk 1]
```diff
@@ -90,6 +90,11 @@
     padding: 0 $spacing-03;
   }
 
+  .#{$prefix}--snippet--single .#{$prefix}--snippet-container,
+  .#{$prefix}--snippet--multi .#{$prefix}--snippet-container {
+    @include focus-outline('reset');
+  }
+
```
Comment (emyarod) [inline packages/components/src/components/code-snippet/_code-snippet.scss]: this isn't needed if we are removing the `tabIndex={0}` from the element right?

**Review by emyarod — COMMENTED** [review]
we can remove the `tabindex` attributes from the vanilla docs so it will show up correctly there too

**Review by laurenmrice — CHANGES_REQUESTED** [review]
- The show more button needs to get a 2px focus border instead of 3px.

- Also I noticed show more ghost button text is mono and needs to be sans `body-short-01`

**Comment by abbeyhrt** [comment]
@tw15egan unfortunately the single line code snippets need to receive focus since they are scrollable. Without it, a keyboard user doesn't have visual access to the whole snippet. This change definitely works for the multi-line snippets though!

**Comment by tw15egan** [comment]
@abbeyhrt hmm, makes sense... Not sure what we want to do in regards to that focused state then.

**Comment by tw15egan** [comment]
Looking at a few other design systems, and none of their code components receive focus 🤔. The focus goes straight to the copy button. We might just have to live with the browser focus if we want to keep that functionality

**Comment by laurenmrice** [comment]
is there no way around creating a focus that has to go outside to accommodate the horizontal scrollbar?  i guess if anything if its only for single line in some browsers we could leave it for now.

**Comment by abbeyhrt** [comment]
@tw15egan I do think the browser focus being misshapen looks less weird than our focus styles misshapen. It seems like this 16px padding-bottom is making the focus the way it is and it seems to be there to vertically center the text, is there a way to vertically center the text without that padding? 
Screen Shot 2020-01-29 at 11 59 44 AM

**Comment by tw15egan** [comment]
This is just one of those components that tends to look different in various browsers/os due to the way the system handles scrolling, i.e 
Screen Shot 2020-01-29 at 10 05 09 AM

I'm hesitant to change the padding / internal structure, which could cause more issues ([List of issues](https://github.com/carbon-design-system/carbon/issues?utf8=%E2%9C%93&q=is%3Aissue+snippet+scroll))  with misalignment in browsers, just to get a perfect focus state. It seems like more work than the perceived benefit.  

I'm good with closing this and the original focus issue with a won't fix tag.

**Comment by abbeyhrt** [comment]
@tw15egan that's totally fair, that list of issues hurts my head haha. I'm good with closing this and keeping the original focus! 👍

**Comment by laurenmrice** [comment]
Can we still keep the changes you did for the multiline code snippet since we don't need it on that variant? I am fine with keeping the single line as it was if there isn't an easy fix for that.

**Comment by tw15egan** [comment]
Just removing focus? Yeah we can add that into that variant

**Comment by tw15egan** [comment]
@laurenmrice just pushed some updates, let me know if this works 👍

**Review by laurenmrice — APPROVED** [review]
yes looks great ! thank you 🙌🏻

**Review by abbeyhrt — APPROVED** [review]
LGTM! 🎉

**Comment by tw15egan** [comment]
@emyarod updated, if you want to take a 2nd look

**Review by emyarod — APPROVED** [review]
looks good to me, I kinda want to keep the ticket open to look a bit more into the single line snippet scroll stuff since I think there's a solution for it

**Comment by tw15egan** [comment]
Go for it! I'll detach the issue from this PR and just leave a ref.

**Pull request merged**